### PR TITLE
[Retiarii] Serializer and experiment status fixes

### PR DESCRIPTION
**Pull request by ultmaster**
This PR fixes:

* Support function object in serializer.
* Show no more trial status when strategy exits.

Will defer the error handling of serializer to follow-up PRs (as it's still too difficult/messy for me). Therefore, serializer is still not confident whether it can correctly handle one case or not.

### CODE REVIEW

File: nni/retiarii/experiment/pytorch.py [hunk 1]
```diff
@@ -131,6 +137,13 @@ def start(self, port: int = 8080, debug: bool = False) -> None:
     def _create_dispatcher(self):
         return self._dispatcher
 
+    def _strategy_monitor(self):
+        while True:
+            time.sleep(2)
+            if not self._strategy_thread.is_alive():
+                self._dispatcher.mark_experiment_as_ending()
+                break
```
Comment (QuanluZhang) [inline nni/retiarii/experiment/pytorch.py:145]: why not directly use `self._strategy_thread.join()` here?

File: nni/retiarii/utils.py [hunk 2]
```diff
@@ -27,11 +28,12 @@ def uid(namespace: str = 'default') -> int:
...
 
 
-def get_module_name(cls):
-    module_name = cls.__module__
+def get_module_name(cls_or_func):
+    module_name = cls_or_func.__module__
     if module_name == '__main__':
         # infer the module name with inspect
         for frm in inspect.stack():
+            print(inspect.getmodule(frm[0]).__name__)
```
Comment (QuanluZhang) [inline nni/retiarii/utils.py:36]: remove this print?

Comment (ultmaster) [reply nni/retiarii/utils.py:36]: Nice catch

File: nni/retiarii/experiment/pytorch.py (hunk 1 above)
Comment (ultmaster) [reply nni/retiarii/experiment/pytorch.py]: Make sense to me.

**Review by QuanluZhang — APPROVED** [review]

**Review by liuzhe-lz — APPROVED** [review]

**Pull request merged**